resources/js/features/games/hooks/useAllMetaRowElements.ts

```diff
@@ -0,0 +1,279 @@
+import { useMemo } from 'react';
+import { route } from 'ziggy-js';
+
+import { extractAndProcessHubMetadata } from '../utils/extractAndProcessHubMetadata';
+
+export function useAllMetaRowElements(
+  game: App.Platform.Data.Game,
+  allGameHubs: App.Platform.Data.GameSet[],
+) {
+  const developerRowElements = useMemo(
+    () =>
+      buildMetaRowElements({
+        hubs: allGameHubs,
+        hubTitleIncludes: ['Developer -', 'Hacker -'],
+        primaryLabel: 'Developer',
+        altLabels: ['Hacker'],
+        fallbackValue: game.developer,
+      }),
+    [allGameHubs, game.developer],
+  );
+
+  const publisherRowElements = useMemo(
+    () =>
+      buildMetaRowElements({
+        hubs: allGameHubs,
+        hubTitleIncludes: ['Publisher - ', 'Hacks -'],
+        primaryLabel: 'Publisher',
+        altLabels: ['Hacks'],
+        fallbackValue: game.publisher,
+      }),
+    [allGameHubs, game.publisher],
+  );
+
+  const genreRowElements = useMemo(
+    () =>
+      buildMetaRowElements({
+        hubs: allGameHubs,
+        hubTitleIncludes: ['Genre -', 'Subgenre -'],
+        primaryLabel: 'Genre',
+        altLabels: ['Subgenre'],
+        fallbackValue: game.genre,
+      }),
+    [allGameHubs, game.genre],
+  );
```

Comment:
Would it make sense to group subgenres under their genre?
![image](https://github.com/user-attachments/assets/944ce0b5-83ad-4527-a78a-7ff677a132af)

would become:
```
Genre:  Action (2D Platforming)
```

Of course, this would only apply when both are linked. I know there's a lot of games that only link the subgenre hub.

Reply:
I think it's a good idea. I've spent some time on this, but it gets really tricky when there are edge cases involved (such as the one you've already mentioned). This is especially complicated if there are multiple genre and subgenre hubs linked to a single game.

Unless you feel strongly, let's push this to a future task for consideration.